strawberryfields/decompositions.py

```diff
@@ -476,6 +476,190 @@ def rectangular_symmetric(V, tol=1e-11):
     return new_tlist, new_diags, None
 
 
+def BS_mat(theta):
+    r"""Beam splitter matrix as implemented in hardware
+    """
+    mat = np.array([[np.cos(theta), 1j*np.sin(theta)], [1j*np.sin(theta), np.cos(theta)]])
+    return mat
+
+def R(phi):
+    r"""Phase as implemented in hardware
+    """
+    mat = np.array([[np.exp(1j*phi), 0], [0, 1]])
+    return mat
+
+def MZ(m, n, phi_int, phi_ext, nmax):
+    r"""One unit as implemented on the chip. 
+    An external phase followed by a beamsplitter followed by 
+    the internal phase followed by another beamsplitter
+    """
```

Comment:
Generally, its best for the first sentence in the docstring to summarise the function concisely, e.g.,

```suggestion
    r"""One Mach-Zehnder unit as implemented on hardware. 
   
    This consists of an external phase, followed by a beamsplitter, followed by 
    the internal phase, followed by another beamsplitter.
    """
```

It could be good in this case to include math rendering in the docstring, either inline or display mode:

```rest
Inline math: :math:`x^2`.

Display math:

.. math::

    BS = \begin{bmatrix}0&1\\1&0\end{matrix}
```

Also, don't forget to add details on the arguments and return type!